specs/language/expressions.tex

```diff
@@ -160,3 +160,87 @@
 overloaded implementation of \texttt{operator[]} (\ref{Overload}).\footnote{HLSL
 does not support the base address of a subscript operator being the expression
 inside the braces, which is valid in C and C++.}
+
+\Sec{Function Calls}{Expr.Post.Call}
+
+\p A function call may be an \textit{ordinary function}, or a \textit{member
+function}. In a function call to an \textit{ordinary function}, the
+\textit{postfix-expression} must be an lvalue that refers to a function. In a
+function call to a \textit{member function}, the \textit{postfix-expression}
+will be an implicit or explicit class member access whose \textit{id-expression}
+is a member function name.
+
+\p When a function is called, each parameter shall be initialized with its
+corresponding argument. The order in which parameters are initialized is
+unspecified. \footnote{Today in DXC targeting DXIL matches the Microsoft C++ ABI
+and evaluates argument expressions right-to-left, while SPIR-V generation
+matches the Itanium ABI evaluating parameters left-to-right. There are good
+arguments for unifying these behaviors, and arguments for keeping them
+different.}
+
+\p If the function is a non-static member function the \texttt{this} argument
+shall be initialized to a reference to the object of the call as if casted by an
+explicit cast expression to an lvalue reference of the type that the function is
+declared as a member of.
+
+\p Parameters are either \textit{input parameters}, \textit{output parameters},
+or \textit{input/output parameters} as denoted in the called function's
+declaration (\ref{Decl.Function}).
+
+\p \textit{Input parameters} are passed by-value into a function. If an argument
+to an \textit{input parameter} is of constant-sized array type, the array is
+copied to a temporary and the temporary value is converted to an address via
+array-to-pointer decay. If an argument is an unsized array type, the array
+lvalue directly decays via array-to-pointer decay. \footnote{This results in
+\textit{input} parameters of unsized arrays being modifiable by a function.}
+
+\p Arguments to \textit{output} and \textit{input/output parameters} must be
+lvalues. \textit{Output parameters} are not initialized prior to the call; they
+are passed as an uninitialized cx-value (\ref{Basic.lval}). An \textit{output
```

Comment:
Nit: Replace `cx-value` with `cxvalue` as defined in `\ref{Basic.lval}`, here and in various places in the document.

Reply:
Done.